src/main/kotlin/org/jetbrains/research/testspark/helpers/psi/kotlin/KotlinPsiHelper.kt

```diff
@@ -0,0 +1,200 @@
+package org.jetbrains.research.testspark.helpers.psi.kotlin
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Caret
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.parentOfType
+import org.jetbrains.kotlin.asJava.toLightClass
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtClassOrObject
+import org.jetbrains.kotlin.psi.KtFunction
+import org.jetbrains.kotlin.psi.KtTypeReference
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.DescriptorToSourceUtils
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+import org.jetbrains.research.testspark.helpers.psi.Language
+import org.jetbrains.research.testspark.helpers.psi.PsiClassWrapper
+import org.jetbrains.research.testspark.helpers.psi.PsiHelper
+import org.jetbrains.research.testspark.helpers.psi.PsiMethodWrapper
+import org.jetbrains.research.testspark.tools.llm.SettingsArguments
+
+class KotlinPsiHelper(private val psiFile: PsiFile) : PsiHelper {
+
+    override val language: Language get() = Language.Kotlin
+
+    private val log = Logger.getInstance(this::class.java)
+
+    override fun generateMethodDescriptor(psiMethod: PsiMethodWrapper): String {
+        val methodDescriptor = psiMethod.methodDescriptor
+        log.info("Method description: $methodDescriptor")
+        return methodDescriptor
+    }
+
+    override fun getSurroundingClass(caretOffset: Int): PsiClassWrapper? {
+        val element = psiFile.findElementAt(caretOffset)
+        val cls = element?.parentOfType<KtClassOrObject>(withSelf = true)
+
+        if (cls != null) {
+            if (cls.name != null) {
```

Comment:
`if(cls?.name !== null)` seems to be an equivalent; at least it is better to make it a single if-statement:
```kotlin
if (cls != null && cls.name != null)
```